pkg/engine/networkinterfaces.go

```diff
@@ -335,17 +339,24 @@ func createAgentVMASNetworkInterface(cs *api.ContainerService, profile *api.Agen
 		}
 		if i == 1 {
 			ipConfig.Primary = to.BoolPtr(true)
+			backendPools := make([]network.BackendAddressPool, 0)
+			if !cs.Properties.OrchestratorProfile.IsPrivateCluster() &&
+				cs.Properties.OrchestratorProfile.KubernetesConfig.LoadBalancerSku == api.StandardLoadBalancerSku {
+				publicLbPool := network.BackendAddressPool{
+					ID: to.StringPtr("[concat(variables('agentLbID'), '/backendAddressPools/', variables('agentLbBackendPoolName'))]"),
+				}
+				backendPools = append(backendPools, publicLbPool)
+			}
```

Comment:
this will also break AKS since AKS provides it's own backend address pool ids. I suggest that we skip this if profile.LoadBalancerBackendAddressPoolIDs != nil

Reply:
If `profile.LoadBalancerBackendAddressPoolIDs != nil`, then we want to add this networkinterface or virtualmachinescalesets to be added to that backendpool right?

Reply:
yes, so we choose between the two set of pools

Reply:
ok thanks for confirming, i will check for `profile.LoadBalancerBackendAddressPoolIDs != nil` first